pkg/arm/data/master-startup.sh

```diff
@@ -422,6 +422,11 @@ cat >/etc/origin/cloudprovider/azure.conf <<'EOF'
 {{ .Derived.CloudProviderConf .ContainerService | String }}
 EOF
 
+
+# when starting node waagent and network utilities goes into race condition.
+# if waagent runs before dns is known to the node we end up with empty string
+while [[ $(hostname -d) == "" ]]; do sleep 1; done
```

Comment:
nitpick: I feel like there's potential this could be stuck in a loop forever, but chances are really slim in reality.